src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+    @abc.abstractmethod 
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        pass
+    
+    @abc.abstractmethod 
+    def assigne(self):
+        """
+        Assigne appropriate python variable to data     
+        """
+        pass
+        
+    @abc.abstractmethod 
+    def getter(self):
+        """
+        Return appropriate python variable to data     
+        """
+        pass
+        
+    @property
+    def value(self):
+        return self.getter()
+        
+class Int(BaseData):
+    """
+    Integer
+    """
+    def __init__(self, integer=None):
+        self.integer = integer
+        """value"""
+    
+    def to_string(self, value):
+        """Presentation of type in json yaml"""
+        return str(value)
+      
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        return  self.integer is not None
+
+    def assigne(self,  value):
+        """
+        Assigne appropriate python variable to data     
+        """
+        if isinstance(value, BaseData):
+            self.integer = int(value.value) 
+        else:
+            self.integer = int(value) 
+
+    def getter(self):
+        """
+        Return appropriate python variable to data     
+        """
+        return self.integer
+        
+class String(BaseData):
+    """
+    String
+    """
+    def __init__(self, string=None):
+        self.string = string
+        """value"""
+    
+    def to_string(self):
+        """Presentation of type in json yaml"""
+        return self.string
+      
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        return  self.string is not None
+        
+    def assigne(self,  value):
+        """
+        Assigne appropriate python variable to data     
+        """
+        if isinstance(value, BaseData):
+            self.string = str(value.value) 
+        else:
+            self.string = str(value)            
+
+    def getter(self):
+        """
+        Return appropriate python variable to data     
+        """
+        return self.string
+        
+class Float(BaseData):
+    """
+    String
+    """
+    def __init__(self, float=None):
+        self.float = float
+        """value"""
+    
+    def to_string(self):
+        """Presentation of type in json yaml"""
+        return float 
+      
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        return  self.float is not None
+
+    def assigne(self,  value):
+        """
+        Assigne appropriate python variable to data     
+        """
+        if isinstance(value, BaseData):
+            self.float = float(value.value) 
+        else:
+            self.float = float(value)
+
+    def getter(self):
+        """
+        Return appropriate python variable to data     
+        """
+        return self.float
+
+class CompositeData(metaclass=abc.ABCMeta):
```

Comment:
abc.ABCMeta je jen definice že jde o abstraktní typ, takže můžeš používat (a taky to dělám) 

if isinstance(value, CompositeDTT):